### DISCUSSION TITLE
{context: 'client'} 'Invalid Id' with @walletconnect/modal-sign-html (index.js:1 {code: 1, message: 'Invalid Id'})

### DISCUSSION THREAD

**Discussion by devon-n** [WalletConnect/walletconnect-monorepo#3092] [Developer Support]
I'm using WalletConnectModalSign with a react site and am able to get the wallet connected.

Once the wallet is connected the user should be prompted to sign a message.

Few times this has been successful but most of the time it comes back with the error message {context: 'client'} 'Invalid Id' and index.js:1 {code: 1, message: 'Invalid Id'}.

It has worked once only with ids 2 and 43

It has not worked with every other combination I've tried. Here is the code

`const response = await this.connector.request({
      topic: this.session.topic,
      chainId: 'eip155:137',
      request: {
        id: '43',
        jsonrpc: '2.0',
        method: 'personal_sign',
        params: [
          address,
          hashedMessage
        ]
      }
    })`

My understanding is that I should be able to set this id to whatever I want as I am not making batch calls.

Where am I going wrong here?

EDIT 1:

I tried with `import SignClient from '@walletconnect/sign-client'
import { WalletConnectModal } from '@walletconnect/modal'`

Where no request Id is put in the signing request and got the same response.

`const response = await this.connector.request({
      topic: this.session.topic,
      chainId: 'eip155:137',
      request: {
        method: 'personal_sign',
        params: [
          address,
          hashedMessage
        ]
      }
    })`

**Comment by crypblizz8** [comment]
Hey @devon-n, happy to try help. Can you share your whole code block for this + your package versions?

**Reply by devon-n** [reply]
I am using the metamask app to scan the qr code

I cannot share the whole code block but these are the package versions

"@walletconnect/modal": "^2.5.9",
"@walletconnect/sign-client": "^2.9.0",
